### PR TITLE
Fix V8 integration for Reanimated V3

### PR DESCRIPTION
**Pull request by capezzbr**
## Description

This PR is an attempt to fix V8 integration with Reanimated V3. Current issue:

```
react-native/node_modules/react-native-reanimated/Common/cpp/ReanimatedRuntime/ReanimatedRuntime.cpp:40:38: error: use of undeclared identifier 'runtime_'
```

## Changes

Updated `ReanimatedRuntime::make` to accept a new parameter `jsi::Runtime *runtime`, which is necessary to initialize V8

## Test code and steps to reproduce

- Create a new RN app
- Setup the app to have V8 as engine
- Enable the new architecture (might not be necessary)
- Build the Android app
- Confirm the app builds and runs fine with this change

## Checklist

- [ ] Included code example that can be used to test this change
- [ ] Updated TS types
- [ ] Added TS types tests
- [ ] Added unit / integration tests
- [ ] Updated documentation
- [ ] Ensured that CI passes

### CODE REVIEW

**Review by piaskowyk — APPROVED** [review]

File: .github/workflows/build-v8.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+on:
+  pull_request:
+    paths:
+      - 'android/**'
+      - 'Common/**'
+      - 'FabricExample/package.json'
+      - 'Example/package.json'
+  push:
+    branches:
+      - main
```
Comment (tomekzaw) [inline .github/workflows/build-v8.yml:11]: Don't forget to change it to `Reanimated2` when cherry-picking this PR.

File: .github/workflows/build-v8.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    concurrency:
+      group: build_v8-${{ matrix.working-directory }}
+      cancel-in-progress: true
+    steps:
+      - name: Check out
+        uses: actions/checkout@v3
+        with:
+          path: 'reanimated_repo'
+      - name: Create react app
+        run: npx react-native init app --version ${{ matrix.working-directory }}
```
Comment (tomekzaw) [inline .github/workflows/build-v8.yml:29]:
```suggestion
        run: npx react-native init app --version ${{ env.REACT_NATIVE_VERSION }}
```

File: .github/workflows/build-v8.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      fail-fast: false
+    concurrency:
+      group: build_v8-${{ matrix.working-directory }}
+      cancel-in-progress: true
+    steps:
+      - name: Check out
+        uses: actions/checkout@v3
+        with:
+          path: 'reanimated_repo'
+      - name: Create react app
```
Comment (tomekzaw) [inline .github/workflows/build-v8.yml:28]:
```suggestion
      - name: Create React Native app
```

File: .github/workflows/helper/configureV8.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    return console.log(err);
+  }
+  data = data.replace('enableHermes: true,', 'enableHermes: false,');
+
+  fs.writeFile(buildGradlePath, data, function (err) {
+    if (err) {
+      return console.log(err);
+    }
+  });
+});
```
Comment (tomekzaw) [inline .github/workflows/helper/configureV8.js:18]: Can we wrap this in a separate function `patch(path, find, replace)`? This way we can eliminate code duplication. Then we can remove const strings and pass code snippets as inline arguments.

File: Common/cpp/ReanimatedRuntime/ReanimatedRuntime.cpp [hunk 5]
```diff
@@ -20,6 +20,7 @@ using namespace facebook;
 using namespace react;
 
 std::shared_ptr<jsi::Runtime> ReanimatedRuntime::make(
+    jsi::Runtime *reactRuntime,
```
Comment (tomekzaw) [inline Common/cpp/ReanimatedRuntime/ReanimatedRuntime.cpp:23]: #3722 introduces `JSRuntimeHelper` which holds `jsi::Runtime *uiRuntime_;` and `jsi::Runtime *rnRuntime_;`. We could unify this PR with the new naming convention and use `rnRuntime` here as well. 
```suggestion
    jsi::Runtime *rnRuntime,
```

File: .github/workflows/build-v8.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        working-directory: [0.70.4]
+      fail-fast: false
```
Comment (tomekzaw) [inline .github/workflows/build-v8.yml:19]:
```suggestion
    env:
      REACT_NATIVE_VERSION: "0.70.4"
```

**Review by tomekzaw — CHANGES_REQUESTED** [review]

File: .github/workflows/helper/configureV8.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  import com.facebook.react.modules.systeminfo.AndroidInfoHelpers;
+  import io.csie.kudo.reactnative.v8.executor.V8ExecutorFactory;
+  `
+);
+
+patchFile(
+  mainApplicationPath,
+  'protected String getJSMainModuleName() {',
+  `
+  protected JavaScriptExecutorFactory getJavaScriptExecutorFactory() {
```
Comment (tomekzaw) [inline .github/workflows/helper/configureV8.js:40]:
```suggestion
  `protected JavaScriptExecutorFactory getJavaScriptExecutorFactory() {
```

File: .github/workflows/helper/configureV8.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+patchFile(
+  'app/android/app/build.gradle',
+  'enableHermes: true,',
+  'enableHermes: false,'
+);
+
+const mainApplicationPath =
+  'app/android/app/src/main/java/com/app/MainApplication.java';
+patchFile(
```
Comment (tomekzaw) [inline .github/workflows/helper/configureV8.js:26]: It looks better with a newline here.
```suggestion
const mainApplicationPath =
  'app/android/app/src/main/java/com/app/MainApplication.java';

patchFile(
```

**Review by tomekzaw — APPROVED** [review]

**Pull request merged**